src/services/render/index.tsx

```diff
@@ -71,14 +72,14 @@ export const renderElement = (
       return null;
     }
 
-    if (
-      options.componentRegistry &&
-      options.componentRegistry[element.namespaceURI] &&
-      options.componentRegistry[element.namespaceURI][element.localName]
-    ) {
-      const Component =
-        options.componentRegistry[element.namespaceURI][element.localName];
+    const Component: HvComponent | undefined = options.componentRegistry
+      ? options.componentRegistry.getComponent(
+          element.namespaceURI,
+          element.localName,
+        )
+      : undefined;
```

Comment:
nit: use optional chaining syntax
```suggestion
    const Component: HvComponent | undefined = options.componentRegistry?.getComponent(
        element.namespaceURI,
        element.localName,
      );
```